roles/validate_inventory/tasks/ai.yml

```diff
@@ -28,3 +28,56 @@
     fail_msg: "{{ item }} is not within the machine network!"
   when: vip_dhcp_allocation == false
   loop: "{{ groups['masters'] + (groups['workers'] | default([])) }}"  # This should not include day2_workers as they can be RWNs
+
+- name: Validate extra VIPs for dualstack
+  when:
+    - extra_api_vip is defined
+    - extra_api_vip | length > 0
+    - extra_ingress_vip is defined
+    - extra_ingress_vip | length > 0
+  block:
+    - name: Assert that Openshift version is supported for dualstack VIPs
+      assert:
+        that:
+          - openshift_full_version is version('4.12', '>=')
+        fail_msg: "openshift_full_version  must be >= 4.12. to support dualstack VIPs"
+
```

Comment:
Can you also add a test that `(extra_machine_networks | length) > 0`
It will be a good validation as having it empty wouldn't be valid anyway and also stops a failure case of the following tests.

Reply:
sure, done